components/raftstore/src/store/peer.rs

```diff
@@ -2786,6 +2803,53 @@ where
         fail_point!("after_send_to_apply_1003", self.peer_id() == 1003, |_| {});
     }
 
+    /// Check long uncommitted proposals and log some info to help find why.
+    pub fn check_long_uncommitted_proposals<T>(&mut self, ctx: &mut PollContext<EK, ER, T>) {
+        if self.has_long_uncommitted_proposals(ctx) {
+            let status = self.raft_group.status();
+            let mut buffer: Vec<(u64, u64, u64)> = Vec::new();
```

Comment:
Does it need to print all the progress? I think just a few sample ones are enough. I noticed this log is too long to read in test environment.

Reply:
In test environment, it just prints the whole Progress object, which contains a vec. Now we just print the last_index and commit_group_id of each peer.